#### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:  

-  первым спортсменом;  
-  вторым спортсменом;  
-  третьим спортсменом.

Это событие может произойти следующими способами:  
$P(B|A_1)$: результате выстрела цель поражена выстрел произведен первым спортсменом $1/3$ так как выстрел произведен определенным спортсменом равны И $P(A_1)$ вероятность попадания = 0.9;  
$P(B|A_2)$: результате выстрела цель поражена выстрел произведен вторым спортсменом $1/3$ так как выстрел произведен определенным спортсменом равны И $P(A_2)$ вероятность попадания = 0.8;  
$P(B|A_3)$: результате выстрела цель поражена выстрел произведен третим спортсменом $1/3$ так как выстрел произведен определенным спортсменом равны И $P(A_3)$ вероятность попадания = 0.6.

Вероятность того, что цель поражена, определяется, как сумма вероятностей этих событий: $P(A)=P(B|A_1)*P(A_1)+P(B|A_2)*P(A_2)+P(B|A_3)*P(A_3)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$  

In [14]:
PBA = 1/3
PB = PBA * 0.9 + PBA * 0.8 + PBA * 0.6
print(f'Вероятность того, что цель поражена P(A) = {round(PB, 4)}')

Вероятность того, что цель поражена P(A) = 0.7667


С помощью формулы Байеса $P(A|B)=\frac{P(B|A)\cdot P(A)}{P(B)}$ оценим вероятность того, что выстрел произведен каждым спортсменом:  
$P(A|B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}$  
$P(A|B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}$  
$P(A|B_3)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$  


In [19]:
PAB1 = PBA * 0.9 / PB
PAB2 = PBA * 0.8 / PB
PAB3 = PBA * 0.6 / PB
print(f'Вероятность того, что выстрел произвёл первый спортсмен: PAB1 = {round(PAB1, 4)} = {round(PAB1 * 100, 2)}%;\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: PAB1 = {round(PAB2, 4)} = {round(PAB2 * 100, 2)}%;\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: PAB1 = {round(PAB3, 4)} = {round(PAB3 * 100, 2)}%.')

Вероятность того, что выстрел произвёл первый спортсмен: PAB1 = 0.3913 = 39.13%;
Вероятность того, что выстрел произвёл второй спортсмен: PAB1 = 0.3478 = 34.78%;
Вероятность того, что выстрел произвёл третий спортсмен: PAB1 = 0.2609 = 26.09%.


### Результат:  

а) Вероятность того, что выстрел произвёл первый спортсмен: 39.13%;  
б) Вероятность того, что выстрел произвёл второй спортсмен: 34.78%;  
в) Вероятность того, что выстрел произвёл третий спортсмен: 26.09%.   